src/chainlink/oracle/ChainlinkOracleImpl.sol

```diff
@@ -0,0 +1,264 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.17;
+
+import {AddressUtils} from "splits-utils/AddressUtils.sol";
+import {AggregatorV3Interface} from "chainlink/interfaces/AggregatorV3Interface.sol";
+import {ConvertedQuotePair, QuotePair, QuoteParams} from "splits-utils/LibQuotes.sol";
+import {QuotePair, QuoteParams} from "splits-utils/LibQuotes.sol";
+import {TokenUtils} from "splits-utils/TokenUtils.sol";
+import {FixedPointMathLib} from "solmate/utils/FixedPointMathLib.sol";
+import {OracleImpl} from "../../OracleImpl.sol";
+import {ChainlinkPairDetails} from "../../libraries/ChainlinkPairDetails.sol";
+
+/// @title Chainlink Oracle Implementation
+/// @author 0xSplits
+/// @notice A clone-implementation of an oracle using Chainlink
+/// @dev This contract uses token = address(0) to refer to ETH.
+contract ChainlinkOracleImpl is OracleImpl {
+    /// -----------------------------------------------------------------------
+    /// libraries
+    /// -----------------------------------------------------------------------
+
+    using AddressUtils for address;
+    using TokenUtils for address;
+    using ChainlinkPairDetails for mapping(address => mapping(address => PairDetail));
+    using FixedPointMathLib for uint256;
+
+    /// -----------------------------------------------------------------------
+    /// errors
+    /// -----------------------------------------------------------------------
+
+    error InvalidPair_FeedNotSet(QuotePair qp);
+    error StalePrice(AggregatorV3Interface feed, uint256 timestamp);
+    error NegativePrice(AggregatorV3Interface feed, int256 price);
+
+    /// -----------------------------------------------------------------------
+    /// structs
+    /// -----------------------------------------------------------------------
+
+    struct InitParams {
+        address owner;
+        bool paused;
+        SetPairDetailParams[] pairDetails;
+    }
+
+    struct SetPairDetailParams {
+        QuotePair quotePair;
+        PairDetail pairDetail;
+    }
+
+    struct PairDetail {
+        // encoded feed[]
+        bytes path;
```

Comment:
remind me the issue here -- you were saying solc can't handle using `Feed[]` here?

Reply:
yup, you can't load a dynamic array into memory

Reply:
oh that -- yeah it just makes `get` and `set` a little messier but not impossible. Going to sleep on it, but if this is the way you prefer to do it that's probably fine. How do you view the trade-off here in your mind?

Reply:
believe it will get really messy to do this the other way and I find the current solution pretty simple in comparison